crates/wysiwyg/src/composer_model/mentions.rs

```diff
@@ -77,7 +77,16 @@ where
         let (start, end) = self.safe_selection();
         let range = self.state.dom.find_range(start, end);
 
-        let new_node = DomNode::new_mention(url, text, attributes);
+        // use the display text or the presence of a `data-mention-type => at-room` attribute to decide the mention type
+        let new_node = if text == "@room".into()
+            || attributes.iter().any(|(k, v)| {
+                k == &S::from("data-mention-type") && v == &S::from("at-room")
```

Comment:
I believe it's an `@room` mention if and only if there is an occurrence of `@room` in the text making this condition redundant

Reply:
Great - I wanted to think that was the case but wasn't sure. Updated in